Effects/Parallax/source/PlayState.hx

```diff
@@ -0,0 +1,230 @@
+		add(_ground);
+		
+		// our guy for the player to move
+		_guy = new FlxSprite(0, 0, "assets/guy.png");
+		_guy.x = ((FlxG.width * 4) / 2) - (_guy.width / 2);
+		_guy.y = FlxG.height - 72 - _guy.height;
+		_baseY = _guy.y;
+		add(_guy);
+		
+		
+		var txtInst:FlxText = new FlxText(0, FlxG.height - 16, FlxG.width, "Left/Right to Move");
+		txtInst.alignment = FlxTextAlign.CENTER;
+		txtInst.setBorderStyle(FlxTextBorderStyle.SHADOW, 0xff333333, 1, 1);
+		txtInst.scrollFactor.set();
+		add(txtInst);
+		
+		// we're going to have some snow or ash flakes drifting down at different 'levels'. We need a lot of them for the effect to work nicely
+		_flakes = new FlxTypedGroup<Flake>();
+		add(_flakes);
+		
+		for (i in 0...1000)
+		{
+			_flakes.add(new Flake(i % 10));
+		}
+		
+		// Set up our camera to follow the guy, and stay within the confines of our 'world'
+		FlxG.camera.follow(_guy);
+		FlxG.camera.setScrollBounds(0, FlxG.width * 4, 0, FlxG.height);
+		
+		// add some tweens to fade the player in and out and to make him 'float' up and down
+		FlxTween.num(0, 1, 2, { ease:FlxEase.sineInOut, type:FlxTween.PINGPONG }, guyFloat);
+		FlxTween.num(.4, 1, 3, { ease:FlxEase.sineInOut, type:FlxTween.PINGPONG }, guyFade);
+		
+		#if (mobile)
+		_vPad = new FlxVirtualPad(FlxDPadMode.LEFT_RIGHT, FlxActionMode.NONE);
+		
+		add(_vPad);
+		#end
+		
+		super.create();
+	}
+	
+	/**
+	 * This function spawns a new Tileblock of clouds which will be positioned near the top of the screen
+	 */
+	private function spawnCloud(Pos:Int):FlxTileblock
+	{
+		var rand:FlxRandom = new FlxRandom();
+		var clouds:FlxTileblock = new FlxTileblock(0, 0, Math.ceil(FlxG.width * 4), 64);
+		clouds.x += -8 + Math.floor(rand.float( 1, 8) * 4);
+		clouds.y += -8 +  Math.floor(rand.float( 1, 8) * Pos);
+		clouds.loadTiles(bakeColors(FlxColor.WHITE.getDarkened(.6 - ( (Pos * .1))),"assets/clouds.png", (1 - (.2 + (Pos * .1) * .5))), 64, 64, Pos * 5);
+		clouds.scrollFactor.set(.2 + (Pos * .1) + .05, 0);
+		return clouds;
+	}
+	
+	/**
+	 * This function generates and returns a new FlxTileblock using our mountain sprites.
+	 */
+	private function spawnMountain(Pos:Int):FlxTileblock
+	{
+		var mountain:FlxTileblock = new FlxTileblock(0, FlxG.height - (180 + ((5 - Pos) * 16)) , Math.ceil(FlxG.width * 4), 116);
+		mountain.loadTiles(bakeColors(FlxColor.WHITE.getDarkened(1 - (.2 + (Pos * .1))), "assets/mountains.png"), 256, 116, 0);
+		mountain.scrollFactor.set(.2 + (Pos * .1), 0);
+		return mountain;
+	}
+	
+	
+	private function guyFloat(Amt:Float):Void
+	{
+		_guy.y = _baseY - Amt * 10;
+		
+	}
+	private function guyFade(Amt:Float):Void
+	{
+		_guy.alpha = Amt;
+		
+	}
+	/**
+	 * this is just logic to make the guy move when the player presses left/right and to keep him within the 'world'
+	 */
+	private function movement():Void
+	{
+		var _left:Bool = false;
+		var _right:Bool = false;
+		#if (mobile)		
+		_left = _vPad.buttonLeft.status == FlxButton.PRESSED;
```

Comment:
Should probably be using `FlxButton.pressed` here (`FlxButton` implements `IFlxInput` on dev).